### ISSUE TITLE
Programmatically Consumable Metadata about clients

### ISSUE THREAD

**Issue by elharo** [googleapis/google-cloud-java#3350]
Cloud Tools for Java has been maintaining a JSON file with info about the client libraries that we can consume in our products:

https://github.com/GoogleCloudPlatform/appengine-plugins-core/blob/master/src/main/resources/com/google/cloud/tools/libraries/libraries.json

We also have Java code for consuming and validating this file.

Any interest in moving this into google-cloud-java so everyone can use it and it stays in sync with releases?

**Comment by andreamlin** [comment]
Are the existing pom.xmls sufficient?

**Comment by garrettjonesgoogle** [comment]
We definitely wouldn't want to manually maintain that - it looks like it reflects data that is present in various places:

1. versions.txt (maven coordinates)
2. Derivable from versions in versions.txt (launchStage)
3. service config (name, serviceName)
4. README.md files (transport)
5. ??? (serviceRoles)

If your concern is the effort to manually maintain it yourself, then I can definitely see why you want to figure out an alternative. Maybe we should figure out how to make all of the datapoints programmatically accessible, so that it is easy to write a script to generate this. It wouldn't necessarily have to live in google-cloud-java; another repo could be set up to automatically regenerate when a new version of google-cloud-java is released. 

TL;DR I suggest we step back and design a solution instead of just moving it.

**Comment by sduskis** [comment]
This is something to work on offline if this is still relevant.
